tests/integration/aliases/adopt-into-component/step1/src/main/java/com/pulumi/example/Main.java

```diff
@@ -0,0 +1,104 @@
+package com.pulumi.example;
+
+import com.pulumi.Pulumi;
+import com.pulumi.core.Output;
+import com.pulumi.resources.ComponentResource;
+import com.pulumi.resources.ComponentResourceOptions;
+import com.pulumi.core.Alias;
+
+import java.util.stream.Collectors;
+import java.util.Objects;
+import java.util.List;
+
+class Resource extends ComponentResource {
+    public Resource(String name, ComponentResourceOptions options) {
+        super("my:module:Resource", name, options);
+    }
+}
+
+// Scenario 1: adopt a resource in to a component. The component author is the same as the component user, and changes
+// the component to be able to adopt the resource that was previously defined separately.
+class Component1 extends ComponentResource {
+    public Component1(String name, String stackName, String projectName, ComponentResourceOptions options) {
+        super("my:module:Component", name, options);
+
+        var adoptedResourceType = "my:module:Resource";
+        var adoptedResourceName = "res";
+
+        // The resource creation was moved from top level to inside the component.
+        var adoptedResource = new Resource(
+                name + "-child",
+                ComponentResourceOptions.builder()
+                    // With a new parent
+                    .parent(this)
+                    // But with an alias provided based on knowing where the resource existing before - in this case at top
+                    // level.  We use an absolute URN instead of a relative `Alias` because we are referencing a fixed resource
+                    // that was in some arbitrary other location in the hierarchy prior to being adopted into this component.
+                    .aliases(Alias.withUrn(String.format("urn:pulumi:%s::%s::%s::%s", stackName, projectName, adoptedResourceType, adoptedResourceName)))
```

Comment:
Is there a better way of doing this in Java? I mean something like `Pulumi.Urn.Create` in C#:

```c#
new ComponentResourceOptions
{
    Parent = this,
    Aliases = { Pulumi.Urn.Create("res2", "my:module:Resource").Apply(urn => new Alias { Urn = urn }) },
});
```

Reply:
`com.pulumi.core.internal.Urn` should be useful, might require small usability changes

Reply:
👍 I saw that, but it is annotated with `@InternalUse`. 
Can we leave the test as is and refine it when some more convenient approach is there (e.g. expose `Urn`). I can create an issue if there aren't any.